### ISSUE TITLE
Is it possible to start notepad from a service in the logged on user?

### ISSUE THREAD

**Issue by ComptonAlvaro** [dotnet/runtime#80050]
I have a gRPC service that is hosted in a ASP Core application. This application is installed in as windows service.

I have a user A in the computer, that log on automatically when the computer starts.

The gRPC service receive commands from a remote client, for example, open the notedpad or another UI application. I would like to open this applications in the logged on user.

However, when I send the command to open the notepad, the application is not opened.

Instead of install the ASP core application as windows service, I have tried to create a scheduled task that starts when the user log on. In this case, it works. The problem is that the application opens a log windows, and I would like to avoid this.

So to avoid to open the log window, I configure the scheduled task to start no matter if the user log on or not. In this case the log window of the ASP application is not opened, but then the notepad doesn't open. The behavior is the same than if I install my ASP application as windows service.

So my question is: is there some way to open the notepad or any other UI application in the logged user from a windows service?

Thanks.

**Comment by danmoseley** [comment]
This is not really a .NET question but really about Windows security and desktop model. You might get a good answer sooner asking on stackoverflow.

**Comment by ComptonAlvaro** [comment]
Yes, I know that perhaps it is more related with Windows than .NET, but I was wondering if it would be some way to do it with C# directly.

For example i know that in C it is possible with the method [CreatedProcessAsUserA()](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessasusera), so I was wondering if in .NET there would be something similiar like this.

**Comment by fbrosseau** [comment]
Even CreateProcessAsUser / WithToken would not change the outcome - it needs to happen from within an interactive login session. Neither the Windows Service nor the background scheduled task are from an interactive login session.

If all that prevents you from using a regular scheduled task is the fact you have a console showing up for your program and you do not want that, consider changing the subsystem from Console to Windows (or in csproj terms - a OutputType of WinExe).

If this is not enough, then you will probably need another layer of IPC. Right now you have this which does not work:

```
|------ non-interactive --------------------| |--- interactive -|
<exterior> -> always-running aspnet service -> desktop programs

You may need an extra layer:

|------ non-interactive --------------------| |--- interactive ----------------|
<exterior> -> always-running aspnet service -> local service -> desktop programs
```

It really only depends on what your specific needs are, but yes this is entirely about Windows details, and not about dotnet. If you need tight control about process creation (such as inheriting specific handles), then you may need to PInvoke one of the CreateProcess* APIs, but otherwise this is not about dotnet.
